pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -174,6 +168,32 @@ func (l *LibvirtDomainManager) SyncVMSecret(vm *v1.VirtualMachine, usageType str
 	return nil
 }
 
+// All local environment setup that needs to occur before VM starts
+// can be done in this function. This includes things like...
+//
+// - storage prep
+// - network prep
+// - cloud-init
+func (l *LibvirtDomainManager) preStartHook(vm *v1.VirtualMachine, domain *api.Domain) error {
```

Comment:
@vladikr This is an entry point you can use for setting up anything in the environment that needs to be setup for networking. It occurs right before the VM starts.